builder/oracle/classic/step_snapshot.go

```diff
@@ -0,0 +1,46 @@
+package classic
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/hashicorp/go-oracle-terraform/compute"
+	"github.com/hashicorp/packer/helper/multistep"
+	"github.com/hashicorp/packer/packer"
+)
+
+type stepSnapshot struct{}
+
+func (s *stepSnapshot) Run(_ context.Context, state multistep.StateBag) multistep.StepAction {
+	// get variables from state
+	ui := state.Get("ui").(packer.Ui)
+	ui.Say("Creating Snapshot...")
+	config := state.Get("config").(*Config)
+	client := state.Get("client").(*compute.ComputeClient)
+	instanceID := state.Get("instance_id").(string)
+
+	// get instances client
+	snapshotClient := client.Snapshots()
+
+	// Instances Input
+	snapshotInput := &compute.CreateSnapshotInput{
+		Instance:     fmt.Sprintf("%s/%s", config.ImageName, instanceID),
+		MachineImage: config.ImageName,
+	}
+
+	snap, err := snapshotClient.CreateSnapshot(snapshotInput)
+	if err != nil {
+		err = fmt.Errorf("Problem creating snapshot: %s", err)
+		ui.Error(err.Error())
+		state.Put("error", err)
+		return multistep.ActionHalt
+	}
+
+	state.Put("snapshot", snap)
+	ui.Message(fmt.Sprintf("Created snapshot: %s.", snap.Name))
+	return multistep.ActionContinue
+}
+
+func (s *stepSnapshot) Cleanup(state multistep.StateBag) {
+	// Nothing to do
```

Comment:
The Snapshot resource can be removed during cleanup.  Note: [deleting the Snapshop](https://docs.oracle.com/en/cloud/iaas/compute-iaas-cloud/stcsa/op-snapshot-%7Bname%7D-delete.html) resource only deletes the snapshop metadata - the [MachineImage](https://docs.oracle.com/en/cloud/iaas/compute-iaas-cloud/stcsa/api-MachineImages.html) resource that created as a side effect of creating snapshot is not removed, and the actual image file in the object storage is also retained.

Reply:
So I should delete the snapshot resource, but not the machine image?  Or is it duplicated when I add it to the image list, and I need to delete both?

Reply:
Just delete the snapshot.